machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -158,8 +158,79 @@ type MachineConfigNodeStatus struct {
 	// +kubebuilder:validation:MaxItems=32
 	// +optional
 	IrreconcilableChanges []IrreconcilableChangeDiff `json:"irreconcilableChanges,omitempty"`
+	// internalReleaseImage describes the status of the release payloads stored in the node.
+	// When specified, an internalReleaseImage custom resource exists on the cluster, and the specified images will be made available on the control plane nodes.
+	// This field will reflect the actual on-disk state of those release images.
+	// +openshift:enable:FeatureGate=NoRegistryClusterOperations
+	// +optional
+	InternalReleaseImage MachineConfigNodeStatusInternalReleaseImage `json:"internalReleaseImage,omitzero,omitempty"`
+}
+
+// MachineConfigNodeStatusInternalReleaseImage holds information about the current and discovered release bundles for the observed machine
+// config node.
+type MachineConfigNodeStatusInternalReleaseImage struct {
+	// releases is a list of the release bundles currently owned and managed by the
+	// cluster, indicating that their images can be safely pulled by any cluster entity
+	// requiring them.
+	// Entries must be unique, keyed on the name field.
+	// This field can contain between 1 and 5 entries.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	// +required
+	Releases []MachineConfigNodeStatusInternalReleaseImageRef `json:"releases,omitempty"`
 }
 
+// MachineConfigNodeStatusInternalReleaseImageRef is used to provide a more detailed reference for
+// a release bundle.
+type MachineConfigNodeStatusInternalReleaseImageRef struct {
+	// conditions represent the observations of an internal release image current state. Valid types are:
+	// Mounted, Installing, Available, Removing and Degraded.
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// name indicates the desired release bundle identifier. This field is required and must be between 1 and 64 characters long.
```

Comment:
What is the format of the release bundle identifier? Is there a strict format that is followed? How did we conclude 64 chars?

Reply:
The [format proposed in the EP](https://github.com/openshift/enhancements/pull/1821/files#diff-7b20b50dcbb0b982828fe87930f63c142457b54311f85d0a73468c86ee2165c1R437) is `ocp-release-bundle-<VER>-<ARCH>`. A len of 64 chars appeared to be a reasonable length to cover also more extreme cases - ie when using a CI stream, like `ocp-release-bundle-ocp-release-bundle-4.20.0-0.nightly-2025-11-06-091216` (in the production case I do not expect to have such lengthy identifiers)

Reply:
>Since this is a format that we have defined, how would you feel about actually adding a validation to this that limits the characters?

Sure, adding a CEL validation for that, please let me know if it's good enough

>     ocp-release-bundle-ocp-release-bundle-4.20.0-0.nightly-2025-11-06-091216
> This is 72 characters? Maybe we want to stick with the DNS subdomain length which is 253 chars?

oops, it looks like a typo. The eact string would be something like `ocp-release-bundle-4.20.0-0.nightly-2025-11-06-091216` - which is 54 chars len, so a max len of 64 seems to be enough